src/bindings/python/src/openvino/frontend/tensorflow/node_decoder.py

```diff
@@ -0,0 +1,146 @@
+    def get_variable(operation):
+        tf_graph = operation.graph
+        if not hasattr(tf_graph, "captures"):
+            return None
+        for var_tensor, op_tensor in tf_graph.captures:
+            if operation.outputs[0].name == op_tensor.name:
+                resource_name = var_tensor._name
+                for variable_value in operation.graph.variables:
+                    if variable_value.name == resource_name:
+                        return variable_value
+                return None
+        return None
+
+    def get_attribute(self, name):
+        if name == "shape" or name == "_output_shapes":
+            if self.m_operation.node_def.attr["shape"].shape.unknown_rank:
+                return OVAny(PartialShape.dynamic())
+            shape_dims = self.m_operation.node_def.attr["shape"].shape.dim
+            shape = [dim.size for dim in shape_dims]
+            type_num = self.m_operation.node_def.attr["dtype"].type
+            if type_num is not None and tf.dtypes.DType(type_num).name == "resource":
+                if self.m_inner_graph:
+                    return OVAny(PartialShape.dynamic())
+                variable_value = TFGraphNodeDecoder.get_variable(self.m_operation)
+                return OVAny(PartialShape(list(variable_value.shape)))
+            return OVAny(PartialShape(shape))
+        if name == "dtype":
+            type_num = self.m_operation.node_def.attr["dtype"].type
+            if tf.dtypes.DType(type_num).name == "resource":
+                if not self.m_inner_graph:
+                    variable_value = TFGraphNodeDecoder.get_variable(self.m_operation)
+                    return OVAny(tf_type_to_ov_type(variable_value.dtype))
+                else:
+                    return OVAny(Type.undefined)
+            return OVAny(tf_type_to_ov_type(type_num))
+
+        if name == "value":
+            if self.m_parsed_content.size == 1:
+                return OVAny(Tensor(np.array([self.m_parsed_content]), shape=[]))
+            ov_tensor = Tensor(self.m_parsed_content, shared_memory=True)
+            ov_tensor = OVAny(ov_tensor)
+            return ov_tensor
+        attr_value = self.m_operation.node_def.attr[name]
+
+        return tf_attr_to_ov(attr_value)
+
+    def get_input_size(self) -> int:
+        return len(self.m_operation.inputs)
+
+    def get_input_node_name(self, input_port_idx):
+        return self.m_operation.inputs[input_port_idx].op.name
+
+    def get_input_node_name_output_port_index(self, input_port_idx):
+        tensor_name = self.m_operation.inputs[input_port_idx].name
+        if ":" in tensor_name:
+            port_idx_str = tensor_name[tensor_name.rfind(":") + 1:len(tensor_name)]
+            if port_idx_str.isdigit():
+                return int(port_idx_str)
+            else:
+                return 0
+        return 0
+
+    def get_input_node_name_output_port_name(self, input_port_idx):
+        tensor_name = self.m_operation.inputs[input_port_idx].name
```

Comment:
please check that `input_port_idx` is not out of range

Reply:
Done.